### PR TITLE
Fix/create separate install target

### LINKED ISSUE #50: Create separate install target
If the install directory is deleted or modified after an initial build and install, one cannot simply re-install the built libraries since there is no install target. I had to run `make clean` and then rebuild everything in order to get it installed again. It would be nice to a have a separate install target so that one could simply run `make install` and the libraries would be installed without rebuilding them unnecessarily.

### PR DESCRIPTION
**Pull request by ktsai7**
Addressing Issue #50: make install now works
- Added a note in README.md
- Edited docker/Dockerfile to account for the change
- Edited .travis.yml for more general tag substitution

Fix #50

### CODE REVIEW

File: cmake/caliper.cmake [hunk 1]
```diff
@@ -7,6 +7,8 @@ message(STATUS "Building ${CALIPER}")
 ExternalProject_Add(${CALIPER}
  URL ${CALIPER_URL}/${CALIPER_GZ}
  URL_MD5 ${CALIPER_MD5}
+ PREFIX third-party
+ INSTALL_DIR third-party/install
```
Comment (junghans) [inline cmake/caliper.cmake]: I wouldn't installed all of the libraries in the same prefix!

File: cmake/exodus.cmake [hunk 2]
```diff
@@ -20,16 +22,18 @@ ExternalProject_Add(${EXODUS_NAME}
    -DCMAKE_Fortran_FLAGS:STRING=${CMAKE_Fortran_FLAGS}
    -DBUILD_SHARED:BOOL=${BUILD_SHARED_LIBS}
    -DBUILD_TESTING:BOOL=OFF
-   -DNETCDF_NCDUMP:PATH=${CMAKE_INSTALL_PREFIX}/bin/ncdump
-   -DNETCDF_SO_ROOT:PATH=${CMAKE_INSTALL_PREFIX}/lib/shared
-   -DNETCDF_INCLUDE_DIR:PATH=${CMAKE_INSTALL_PREFIX}/include
+   -DNETCDF_NCDUMP:PATH=<INSTALL_DIR>/bin/ncdump
```
Comment (junghans) [inline cmake/exodus.cmake]: You can get the NetCDF prefix with `ExternalProject_get_property(${NETCDF_NAME} INSTALL_DIR)`

**Review by junghans — CHANGES_REQUESTED** [review]

File: cmake/caliper.cmake (hunk 1 above)
Comment (ktsai7) [reply cmake/caliper.cmake]: So something like 
PREFIX project_name
INSTALL_DIR project_name/install
?

Comment (junghans) [reply cmake/caliper.cmake]: You could also just leave it with the default. Or was there a reason to install all in the same prefix?

Comment (ktsai7) [reply cmake/caliper.cmake]: I had it as `PREFIX project_name` but then during build legion cannot find gasnet include dir and conduits...

File: docker/Dockerfile [hunk 3]
```diff
@@ -8,28 +8,40 @@ ARG TRAVIS
...
+          -DCMAKE_INSTALL_PREFIX=/home/flecsi/tpl/install/ \
+          ..
+RUN make -j2
+RUN ccache -s
 
 USER root
-RUN chown -R root:root /usr/local
+RUN make install
 USER flecsi
+RUN ccache -s
```
Comment (junghans) [inline docker/Dockerfile]: A second time?

File: docker/Dockerfile [hunk 4]
```diff
@@ -8,28 +8,40 @@ ARG TRAVIS
...
+WORKDIR build
+RUN ccache -z
+RUN cmake -DBUILD_SHARED_LIBS=ON \
           -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
           -DUSE_SYSTEM_LIBS=${SYSTEM_LIBS} \
-          .. && make -j2;
+          -DGASNet_ROOT_DIR=/home/flecsi/tpl/install \
+          -DCMAKE_INSTALL_PREFIX=/home/flecsi/tpl/install/ \
+          ..
+RUN make -j2
```
Comment (junghans) [inline docker/Dockerfile]: Please also run `male install DESTDIR=$PWD` to test the install.

File: docker/Dockerfile (hunk 3 above)
Comment (ktsai7) [reply docker/Dockerfile]: Ohh it's copied over from the dockerfile from vocta...

File: docker/Dockerfile [hunk 5]
```diff
@@ -8,28 +8,38 @@ ARG TRAVIS
...
           -DUSE_SYSTEM_LIBS=${SYSTEM_LIBS} \
-          .. && make -j2;
+          -DCMAKE_INSTALL_PREFIX:PATH=/home/flecsi/tpl/install/ \
+          ..
+RUN make install -j2
+RUN ccache -s
 
 USER root
-RUN chown -R root:root /usr/local
+RUN make DESTDIR=$PWD install
```
Comment (junghans) [inline docker/Dockerfile]: No, `make install` with`DESTDIR` should be run as normal user and then a normal `make install` without `DESTDIR` as root.

**Review by junghans — CHANGES_REQUESTED** [review]

File: docker/Dockerfile [hunk 6]
```diff
@@ -8,28 +8,41 @@ ARG TRAVIS
...
-    cmake -DBUILD_SHARED_LIBS=ON \
+RUN mkdir build
+RUN mkdir install
+WORKDIR build
+RUN ccache -z
+RUN cmake -DBUILD_SHARED_LIBS=ON \
           -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
           -DUSE_SYSTEM_LIBS=${SYSTEM_LIBS} \
-          .. && make -j2;
+          -DCMAKE_INSTALL_PREFIX:PATH=/home/flecsi/tpl/install/ \
```
Comment (junghans) [inline docker/Dockerfile]: Why are we installing it in `/usr/local` anymore?

Comment (ktsai7) [reply docker/Dockerfile]: I thought this is easier. Was there a particular reason we want to install in usr/local?

Comment (junghans) [reply docker/Dockerfile]: `/usr/local` is in CMake's default search path, so we don't need to specify the path (e.g. in `Legion_ROOT`) in the dependent containers.

**Review by junghans — APPROVED** [review]

**Pull request merged**